Mismatched types in bindings (windows)

Building this on windows generates a lot of mismatched types errors like these for me:
```rs
error[E0308]: mismatched types
   --> src\command.rs:113:44
    |
113 |                 load_op: conv::map_load_op(desc.depthLoadOp),
    |                                            ^^^^^^^^^^^^^^^^ expected `i32`, found `u32`
    |
help: you can convert a `u32` to an `i32` and panic if the converted value doesn't fit
    |
113 |                 load_op: conv::map_load_op(desc.depthLoadOp.try_into().unwrap()),
    |                
```

Removing/changing [src/lib.rs@124](https://github.com/gfx-rs/wgpu-native/blob/636629dcad96a1e450741d61e1d8635cac36785e/src/lib.rs#L124) 
```rs
#[cfg(target_os = "windows")]
pub type EnumConstant = i32;
```
to `u32` fixes this for me, so I'd like to change it.
However I don't know if maybe I'm on some other version of some tool or this might break compilation for anyone else

```
> rustc --version
rustc 1.58.0 (02072b482 2022-01-11)
> cargo version
cargo 1.58.0 (7f08ace4f 2021-11-24)
> rustup default
stable-x86_64-pc-windows-msvc (default)
> bindgen --version
bindgen 0.59.2

```

seems like, this is still an issue when building with `stable-x86_64-pc-windows-gnu` toolchain.
[here my actions run](https://github.com/rajveermalviya/go-webgpu/runs/4936938559)

```
error[E0308]: mismatched types
   --> src\command.rs:113:44
    |
113 |                 load_op: conv::map_load_op(desc.depthLoadOp),
    |                                            ^^^^^^^^^^^^^^^^ expected `u32`, found `i32`
    |
help: you can convert an `i32` to a `u32` and panic if the converted value doesn't fit
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -72,16 +72,27 @@ jobs:
               rm -rf $GLFW_DIR
             make_command: make VERBOSE=1
           - os: windows-2019
-            name: Windows Stable
+            name: Windows Stable (msvc)
             channel: stable
-            build_command: rustup default stable-msvc; cargo clippy
+            set_toolchain: rustup default stable-msvc
+            build_command: cargo clippy
             make_command: make lib-native
             install_deps_command: |
               # choco exit with code 1 after successful install
               choco install -y --force llvm | exit 0
               echo "LIBCLANG_PATH=C:\Program Files\LLVM\lib" >> $GITHUB_ENV
           - os: windows-2019
-            name: Windows Nightly
+            name: Windows Stable (gnu)
+            channel: stable
+            set_toolchain: rustup default stable-gnu
+            build_command: cargo clippy
+            make_command: make lib-native
+            install_deps_command: |
+              # choco exit with code 1 after successful install
+              choco install -y --force llvm | exit 0
+              echo "LIBCLANG_PATH=C:\Program Files\LLVM\lib" >> $GITHUB_ENV
+          - os: windows-2019
+            name: Windows Nightly (msvc)
             channel: nightly
             #build_command: rustup default nightly-msvc; cargo check
             # speed up this build a little
@@ -110,6 +121,8 @@ jobs:
         with:
           toolchain: nightly
           override: true
+      - name: Set toolchain
+        run: ${{ matrix.set_toolchain }}
       - name: Install clippy
         if: contains(matrix.build_command, 'clippy')
         run: rustup component add clippy

--- a/src/conv.rs
+++ b/src/conv.rs
@@ -377,7 +377,9 @@ pub fn map_color(native: &native::WGPUColor) -> wgt::Color {
     }
 }
 
-pub fn map_texture_view_dimension(value: crate::EnumConstant) -> Option<wgt::TextureViewDimension> {
+pub fn map_texture_view_dimension(
+    value: native::WGPUTextureViewDimension,
+) -> Option<wgt::TextureViewDimension> {
     match value {
         native::WGPUTextureViewDimension_1D => Some(wgt::TextureViewDimension::D1),
         native::WGPUTextureViewDimension_2D => Some(wgt::TextureViewDimension::D2),
@@ -389,7 +391,7 @@ pub fn map_texture_view_dimension(value: crate::EnumConstant) -> Option<wgt::Tex
     }
 }
 
-pub fn map_texture_dimension(value: crate::EnumConstant) -> wgt::TextureDimension {
+pub fn map_texture_dimension(value: native::WGPUTextureDimension) -> wgt::TextureDimension {
     match value {
         native::WGPUTextureDimension_1D => wgt::TextureDimension::D1,
         native::WGPUTextureDimension_2D => wgt::TextureDimension::D2,
@@ -398,7 +400,7 @@ pub fn map_texture_dimension(value: crate::EnumConstant) -> wgt::TextureDimensio
     }
 }
 
-pub fn map_texture_format(value: crate::EnumConstant) -> Option<wgt::TextureFormat> {
+pub fn map_texture_format(value: native::WGPUTextureFormat) -> Option<wgt::TextureFormat> {
     // TODO: Add support for BC formats
     match value {
         native::WGPUTextureFormat_R8Unorm => Some(wgt::TextureFormat::R8Unorm),
@@ -444,7 +446,7 @@ pub fn map_texture_format(value: crate::EnumConstant) -> Option<wgt::TextureForm
     }
 }
 
-pub fn to_native_texture_format(rs_type: wgt::TextureFormat) -> crate::EnumConstant {
+pub fn to_native_texture_format(rs_type: wgt::TextureFormat) -> native::WGPUTextureFormat {
     match rs_type {
         wgt::TextureFormat::R8Unorm => native::WGPUTextureFormat_R8Unorm,
         wgt::TextureFormat::R8Snorm => native::WGPUTextureFormat_R8Snorm,

--- a/src/device.rs
+++ b/src/device.rs
@@ -523,7 +523,7 @@ pub unsafe extern "C" fn wgpuBufferMapAsync(
     user_data: *mut u8,
 ) {
     let operation = wgc::resource::BufferMapOperation {
-        host: match mode as crate::EnumConstant {
+        host: match mode as native::WGPUMapMode {
             native::WGPUMapMode_Write => wgc::device::HostMap::Write,
             native::WGPUMapMode_Read => wgc::device::HostMap::Read,
             native::WGPUMapMode_None => panic!("Buffer map mode None is not supported."),

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -120,12 +120,6 @@ macro_rules! follow_chain {
     }}};
 }
 
-#[cfg(all(target_os = "windows", target_env = "msvc"))]
-pub type EnumConstant = i32;
-
-#[cfg(not(all(target_os = "windows", target_env = "msvc")))]
-pub type EnumConstant = u32;
-
 /// Creates a function which maps native constants to wgpu enums.
 /// If an error message is provided, the function will panic if the
 /// input does not match any known variants. Otherwise a Result<T, i32> is returned
@@ -171,7 +165,7 @@ pub type EnumConstant = u32;
 #[macro_export]
 macro_rules! map_enum {
     ($name:ident, $c_name:ident, $rs_type:ty, $($variant:ident),+) => {
-        pub fn $name(value: crate::EnumConstant) -> Result<$rs_type, crate::EnumConstant> {
+        pub fn $name(value: native::$c_name) -> Result<$rs_type, native::$c_name> {
             match value {
                 $(paste::paste!(native::[<$c_name _ $variant>]) => Ok(<$rs_type>::$variant)),+,
                 x => Err(x),
@@ -179,14 +173,14 @@ macro_rules! map_enum {
         }
     };
     ($name:ident, $c_name:ident, $rs_type:ty, $err_msg:literal, $($variant:ident),+) => {
-        pub fn $name(value: crate::EnumConstant) -> $rs_type {
+        pub fn $name(value: native::$c_name) -> $rs_type {
             map_enum!(map_fn, $c_name, $rs_type, $($variant),+);
 
             map_fn(value).expect($err_msg)
         }
     };
     ($name:ident, $c_name:ident, $rs_type:ty, $($native_variant:ident:$variant2:ident),+) => {
-        pub fn $name(value: crate::EnumConstant) -> Result<$rs_type, crate::EnumConstant> {
+        pub fn $name(value: native::$c_name) -> Result<$rs_type, native::$c_name> {
             match value {
                 $(paste::paste!(native::[<$c_name _ $native_variant>]) => Ok(<$rs_type>::$variant2)),+,
                 x => Err(x),
@@ -194,7 +188,7 @@ macro_rules! map_enum {
         }
     };
     ($name:ident, $c_name:ident, $rs_type:ty, $err_msg:literal, $($native_variant:ident:$variant2:ident),+) => {
-        pub fn $name(value: crate::EnumConstant) -> $rs_type {
+        pub fn $name(value: native::$c_name) -> $rs_type {
             map_enum!(map_fn, $c_name, $rs_type, $($native_variant:$variant2),+);
 
             map_fn(value).expect($err_msg)

